src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -223,5 +218,63 @@ public override int GetHashCode(object obj)
 				return base.GetHashCode(obj);
 			}
 		}
+
+		/// <summary>
+		/// A factory method that can be used to supply <see cref="Arguments"/> to <see cref="IKernel.Resolve{T}(Castle.MicroKernel.Arguments)"/>. Creates an <see cref="Arguments"/> instance from an IDictionary.
+		/// </summary>
+		/// <param name="dictionary">Key values are copied into <see cref="Arguments"/> where the values are used as dependencies</param>
+		/// <returns><see cref="Arguments"/></returns>
+		public static Arguments FromDictionary(IDictionary dictionary)
+		{
+			return new Arguments(dictionary);
+		}
+
+		/// <summary>
+		/// A factory method that can be used to supply <see cref="Arguments"/> to <see cref="IKernel.Resolve{T}(Castle.MicroKernel.Arguments)"/>. Creates an <see cref="Arguments"/> instance from an anonymous type.
+		/// </summary>
+		/// <param name="anonymous">Anonymous type where the property values are reflected for values and copied into <see cref="Arguments"/> and then used as dependencies</param>
+		/// <returns><see cref="Arguments"/></returns>
+		public static Arguments FromProperties(object anonymous)
```

Comment:
As we discussed in the thread `FromProperties` works just as well with a POCO as it does an anonymous type, so we should make this factory method clear. I'd change the parameter name to "instance" or "object" or something, and update the XML docs to reflect it works on "public properties of .NET objects including anonymous types".